### DISCUSSION TITLE
Attemting to get neotest working with php / pest adapter

### DISCUSSION THREAD

**Discussion by thyseus** [AstroNvim/astrocommunity#1037] [Q&A]
Hey there, 

i am trying to use neotest with the https://github.com/V13Axel/neotest-pest adapter.

I added this to my user.lua
```
  {
    "nvim-neotest/neotest",
    dependencies = {
      "V13Axel/neotest-pest",
    },
    config = function()
      require("neotest").setup {
        adapters = {
          require "neotest-pest" {
            sail_enabled = true,
          },
        },
      }
    end,
  },
```

community.lua:
```
  { import = "astrocommunity.test.neotest" },
```

and everything *seems* fine (no error messages, lazy installs everything correctly).

But when i try to run a test, it always says "No Tests found". 

I think i am doing something wrong here, could one please help me ? 
Maybe add the correct way to add an adapter to the documentation ? 
Or make the adapters enable/disable-able in the astrocommunity plugin ?

Thanks a lot !

**Comment by H-Hp** [comment]
It seems like you have correctly set up the Neotest and neotest-pest adapter in your Neovim configuration. However, there might be an issue with the configuration or the way the adapter is detecting the tests.

Here are a few things you can try:

1. **Check the adapter configuration**:
   Make sure that the adapter configuration is correct for your project. The `neotest-pest` adapter expects your tests to be located in the `tests` directory by default. If your tests are located elsewhere, you might need to update the configuration accordingly.

   You can try modifying the adapter configuration in your `user.lua` file like this:

   ```lua
   require("neotest").setup {
     adapters = {
       require("neotest-pest")({
         dap = {
           type = "server",
           port = 9000,
           executable = { command = "php", args = { "-dxdebug.mode=debug" } },
           options = {
             source_filetype = "php",
           },
         },
         test_dir = "tests", -- Adjust this if your tests are in a different directory
       }),
     },
   }
   ```

2. **Check if tests are being detected**:
   You can try running the `:Neotest` command and inspect the output. If no tests are listed, it means that the adapter is not detecting your tests correctly.

3. **Check the adapter documentation**:
   Refer to the adapter's documentation or repository for any specific configuration requirements or known issues. The adapter might have additional setup steps or limitations that need to be addressed.

4. **Enable debugging**:
   You can try enabling debugging for Neotest to get more insight into what's happening. Add the following line to your `user.lua` file:

   ```lua
   vim.g.neotest_debug = true
   ```

   This will enable verbose logging, which might help you identify the issue.

5. **Open an issue**:
   If you've tried all the above steps and still can't get it to work, consider opening an issue on the `neotest-pest` adapter repository. Provide detailed information about your setup, the steps you've taken, and any error messages or logs you've encountered. The maintainers or the community might be able to assist you further.

Regarding your suggestion about making adapters enable/disable-able in the AstroNvim community plugin, you can open an issue or a pull request on the AstroNvim repository to discuss this feature request. The maintainers might consider adding this functionality if it aligns with the project's goals and if there is a demand for it.
